tests/bluetooth/iso/modules/iso_broadcast_sink.c

```diff
@@ -0,0 +1,535 @@
+		    info->interval * 5 / 4,
+		   phy2str(info->phy));
+
+	k_sem_give(&sem_per_sync);
+}
+
+static void term_cb(struct bt_le_per_adv_sync *sync,
+			const struct bt_le_per_adv_sync_term_info *info)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+
+	LOG_DBG("PER_ADV_SYNC[%u]: [DEVICE]: %s sync terminated",
+		   bt_le_per_adv_sync_get_index(sync), le_addr);
+
+	per_adv_lost = true;
+	k_sem_give(&sem_per_sync_lost);
+}
+
+static void recv_cb(struct bt_le_per_adv_sync *sync,
+			const struct bt_le_per_adv_sync_recv_info *info, struct net_buf_simple *buf)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+	char data_str[129];
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+	bin2hex(buf->data, buf->len, data_str, sizeof(data_str));
+
+	LOG_DBG("PER_ADV_SYNC[%u]: [DEVICE]: %s, tx_power %i, "
+		   "RSSI %i, CTE %u, data length %u, data: %s",
+		   bt_le_per_adv_sync_get_index(sync), le_addr, info->tx_power, info->rssi,
+		   info->cte_type, buf->len, data_str);
+}
+
+static void biginfo_cb(struct bt_le_per_adv_sync *sync, const struct bt_iso_biginfo *biginfo)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(biginfo->addr, le_addr, sizeof(le_addr));
+
+	LOG_DBG("BIG INFO[%u]: [DEVICE]: %s, sid 0x%02x, "
+		   "num_bis %u, nse %u, interval 0x%04x (%u ms), "
+		   "bn %u, pto %u, irc %u, max_pdu %u, "
+		   "sdu_interval %u us, max_sdu %u, phy %s, "
+		   "%s framing, %sencrypted",
+		   bt_le_per_adv_sync_get_index(sync), le_addr, biginfo->sid, biginfo->num_bis,
+		   biginfo->sub_evt_count, biginfo->iso_interval, (biginfo->iso_interval * 5 / 4),
+		   biginfo->burst_number, biginfo->offset, biginfo->rep_count, biginfo->max_pdu,
+		   biginfo->sdu_interval, biginfo->max_sdu, phy2str(biginfo->phy),
+		   biginfo->framing ? "with" : "without", biginfo->encryption ? "" : "not ");
+
+	k_sem_give(&sem_per_big_info);
+}
+
+static struct bt_le_per_adv_sync_cb sync_callbacks = {
+	.synced = sync_cb,
+	.term = term_cb,
+	.recv = recv_cb,
+	.biginfo = biginfo_cb,
+};
+
+static void iso_recv(struct bt_iso_chan *chan, const struct bt_iso_recv_info *info,
+			 struct net_buf *buf)
+{
+	int ret = 0;
+	uint32_t count = 0;
+	static uint32_t last_count;
+	static uint32_t counts_fail;
+	static uint32_t counts_success;
+
+	if (buf->len == sizeof(count)) {
+		count = sys_get_le32(buf->data);
+	}
+
+	if (!(info->flags & BT_ISO_FLAGS_VALID)) {
+		LOG_ERR("bad frame");
+	}
+
+	/* Wrapping not considered. Overflow will take over a year.*/
+	if (last_count) {
+		if (count != (last_count + 1)) {
+			counts_fail++;
+		} else {
+			counts_success++;
+		}
+	}
+
+	last_count = count;
+
+	if ((count % CONFIG_PRINT_CONN_INTERVAL) == 0) {
+		LOG_INF("RX. Count: %d, Failed: %d, Success: %d", count, counts_fail,
+			   counts_success);
+
+		if ((count/CONFIG_PRINT_CONN_INTERVAL) % 2 == 0) {
+			ret = gpio_pin_set_dt(&led, 1);
+		} else {
+			ret = gpio_pin_set_dt(&led, 0);
+		}
```

Comment:
What is the purpose of the `(count/CONFIG_PRINT_CONN_INTERVAL) % 2` calculation here? 

If you just want to switch it every time it enters here, perhaps something like `gpio_pin_toggle_dt` could just be used?

Reply:
If it is done in this manner, then they toggle on and off in sync. Pimpety!

Reply:
Not sure I understood that :D 

If count == 100 and CONFIG_PRINT_CONN_INTERVAL is 5, then 100/5 = 20 % 2 = 0 and it sets the LED to 1. When count == 101, then it 101 / 5 = 20 % 2 = 0, and stays the same. Same for 102, 103 and 104, and then it switches on 105, and stays there for 106, 107, 108, 109, and switches again on 110. So far so good. 

However, since this is inside a `if ((count % CONFIG_PRINT_CONN_INTERVAL) == 0) {`, then it will only ever enter this on 100, 105, 110, etc. 

So 
```c
	if ((count % CONFIG_PRINT_CONN_INTERVAL) == 0) {
		LOG_INF("RX. Count: %d, Failed: %d, Success: %d", count, counts_fail,
			   counts_success);

		if ((count/CONFIG_PRINT_CONN_INTERVAL) % 2 == 0) {
			ret = gpio_pin_set_dt(&led, 1);
		} else {
			ret = gpio_pin_set_dt(&led, 0);
		}
	}
```

gives the same result as 

```c
	if ((count % CONFIG_PRINT_CONN_INTERVAL) == 0) {
		LOG_INF("RX. Count: %d, Failed: %d, Success: %d", count, counts_fail,
			   counts_success);

		gpio_pin_toggle_dt(&led);
	}
```

does it not?

Reply:
On one kit yes, but if we power on kit A at packet 0 and kit B at packet 101, then A will turn the LED on and B will turn it off.

Reply:
That's a fair point :)

In any case, I'd suggest to use `& 1` instead of `% 2` to check if it is even or odd, as the modulo operator is an expensive operation (even though I assume that any reasonable compile would compile `% 2` to `& 1`). 

It's also OK to leave as is